### PR TITLE
add vcn checkout

### PR DESCRIPTION
**Pull request by jackdebugging**

### CODE REVIEW

File: AffirmSDK.podspec [hunk 1]
```diff
@@ -7,7 +7,7 @@ Pod::Spec.new do |spec|
     spec.license                = { :type => "BSD-3-Clause", :file => "LICENSE.txt" }
     spec.source                 = { :git => "https://github.com/Affirm/affirm-ios-sdk.git", :tag => spec.version }
     spec.source_files           = "AffirmSDK/*.{h,m}"
-    spec.platform               = :ios, "7.0"
```
Comment (hoyelo) [inline AffirmSDK.podspec:10]: @YiJie-Jack Pls confirm with client whether we can change this.

File: AffirmSDK/AffirmCheckoutDelegate.h [hunk 2]
```diff
@@ -26,6 +26,8 @@ NS_ASSUME_NONNULL_BEGIN
 /// For more information about the server integration, see https://docs.affirm.com/v2/api/charges
 - (void)checkout:(AffirmCheckoutViewController *)checkoutVC completedWithToken:(NSString *)checkoutToken;
 
+- (void)vcnCheckout:(AffirmCheckoutViewController *)checkoutVC completedWithCardInfo:(NSString *)cardInfo;
```
Comment (hoyelo) [inline AffirmSDK/AffirmCheckoutDelegate.h:34]: Please add function explanation. @YiJie-Jack

File: AffirmSDK/AffirmCheckoutViewController.m [hunk 3]
```diff
@@ -109,6 +113,41 @@ - (void) prepareForCheckout {
...
+                                                                withString:obj
+                                                                   options:NSLiteralSearch
+                                                                     range:[rawContent rangeOfString:key]];
+        }];
+        
+        NSURLComponents *urlComponents = [[NSURLComponents alloc] initWithString:redirect_url];
+        NSString *baseUrl = [NSString stringWithFormat:@"https://%@", urlComponents.host];
+        
+        if (@available(iOS 9.0, *)) {
+            [self.webView loadData:[rawContent dataUsingEncoding:NSUTF8StringEncoding] MIMEType:@"text/html" characterEncodingName:@"utf-8" baseURL:[NSURL URLWithString: baseUrl]];
```
Comment (hoyelo) [inline AffirmSDK/AffirmCheckoutViewController.m:136]: Remove the space after colon pls, and other places. @YiJie-Jack

File: AffirmSDK/AffirmCheckoutViewController.m [hunk 4]
```diff
@@ -135,15 +174,23 @@ - (void) webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigat
...
+            } else {
+                if([item.name isEqualToString:@"checkout_token"]) {
+                    checkoutData = item.value;
+                    [self.delegate checkout:self completedWithToken:checkoutData];
+                    break;
+                }
             }
         }
-        [self.delegate checkout:self completedWithToken:checkoutToken];
-        [AffirmLogger logEvent:@"Checkout completed" info:@{@"checkout_ari": self.checkoutARI, @"checkout_token_received": checkoutToken != nil ? @"true" : @"false"}];
```
Comment (hoyelo) [inline AffirmSDK/AffirmCheckoutViewController.m:146]: @YiJie-Jack Why did you remove this event?

Comment (YiJie-Jack) [reply AffirmSDK/AffirmCheckoutViewController.m]: i don't remove this event, just modified a variable name 
checkoutToken -> checkoutData

File: AffirmSDK/AffirmCheckoutViewController.m (hunk 3 above)
Comment (jackdebugging) [reply AffirmSDK/AffirmCheckoutViewController.m:136]: fixed

File: AffirmSDK/AffirmCheckoutDelegate.h (hunk 2 above)
Comment (jackdebugging) [reply AffirmSDK/AffirmCheckoutDelegate.h:34]: added
